unit_tests_quick_start_guide.rst

```diff
@@ -0,0 +1,165 @@
+
+Unit tests are tests written to ensure that a section of code (known as ``unit``) meets its design and behaves as intended.
+
+When we say ``unit``, we usually imply a function or method of a class used in a module or plugin.
+
+With unit testing, we check if a function when it gets a certain input, returns an output that we expect.
+
+If a function raises / handles exceptions, this should also be checked.
+
+Ansible uses `pytest <https://docs.pytest.org/en/latest/>`_ as a testing framework.
+
+To learn why use unit tests, refer to the `Unit testing Ansible modules <https://docs.ansible.com/ansible/devel/dev_guide/testing_units_modules.html#why-use-unit-tests>`_ documentation page.
+
+It is a `requirement <https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#requirements-for-collections-to-be-included-in-the-ansible-package>`_ for inclusion of a collection to the Ansible package to have integration and / or unit tests. However, even if your collection already has / is going to have integration tests, it will probably make your code more reliable and bug-introduction sensitive if your modules are also covered by unit tests and vice versa. To learn how to get started with integration tests, see the `Quick-start integration testing guide <integration_tests_quick_start_guide.rst>`_.
+
+.. _Prepare-local-environment:
+
+Prepare local environment
+=========================
+
+Before starting on the unit tests themselves, in order to run them locally, you will need to prepare your environment. To learn how to do it quickly, refer to the `Quick-start development guide <https://github.com/ansible/community-docs/blob/main/create_pr_quick_start_guide.rst#prepare-your-environment>`_.
+
+.. _Determine-if-unit-tests-exists:
+
+Determine if unit tests exist
+=============================
+
+Ansible collection unit tests are located in the ``tests/units`` directory.
+
+The structure of the unit tests matches the structure of the code base, so the tests can reside in the ``tests/units/plugins/modules/`` and ``tests/units/plugins/module_utils`` directories. There can be sub-directories, if, for example, modules are organized by module groups.
+
+When you are thinking of adding unit tests for the module called, say, ``my_module``, to check if they already exist, see if there is a file in the collection's source tree with the path ``tests/units/plugins/modules/test_my_module.py``.
+
+Example of unit tests
+=====================
+
+Let's assume that there is the following function in ``my_module``:
+
+.. code:: python
+
+  def convert_to_supported(val):
+      """Convert unsupported types to appropriate."""
+      if isinstance(val, decimal.Decimal):
+          return float(val)
+
+      elif isinstance(val, datetime.timedelta):
+          return str(val)
+
```

Comment:
Maybe we should also throw an exception here under certain conditions? This would allow us to showcase how we can use pytest to check for exceptions.

```suggestion

    if val == 42:
        raise ValueError("This number is just too cool for us ;)")

```

Reply:
i intentionally didn't provide details because Ansible unit testing doc is pretty descent but it looks good to me:)

Reply:
committed, thanks:)